### PR TITLE
RAG Blueprint: Add second phase ranking with gbdt

### PR DESCRIPTION
**Pull request by thomasht86**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Collected training data with `VespaFeatureCollector`from pyvespa (in [review](https://github.com/vespa-engine/pyvespa/pull/1076)). 
Also collected some metadata-fields. 
Trained lgbm-model and use this for second-phase with new rank-profile. 

Would appreciate if someone could run the steps in the README to verify if all steps are correct. 🙏 

fyi @radu-gheorghe @kaibor737

### CODE REVIEW

File: rag-blueprint/README.md [hunk 1]
```diff
@@ -285,3 +288,106 @@ Intercept                     : -3.5974
...
+2025-06-20 05:34:54,225 - INFO -   nativeRank: 18.3844
+2025-06-20 05:34:54,225 - INFO -   max_chunk_sim_scores: 15.1054
+2025-06-20 05:34:54,225 - INFO -   elementCompleteness(chunks).completeness: 7.0140
+2025-06-20 05:34:54,225 - INFO - Selected 16 features with non-zero importance
+2025-06-20 05:34:54,225 - INFO - Training final model on all data for 21 rounds
+2025-06-20 05:34:54,243 - INFO - Model exported to /Users/thomas/Repos/sample-apps/rag-blueprint/eval/Vespa-training-data_second-phase_20250619_182246_lightgbm_model.json
+2025-06-20 05:34:54,243 - INFO - Training completed successfully!
+```
+
+Great! We now have a trained GBDT model that we will use for our second-phase ranking.
```
[the remark this answers is not in the document]
Comment (andreer) [reply rag-blueprint/README.md:372]: I think a little bit here too doesn't hurt, then later we can add a link to other materials for more details for anyone who encounters this README first

**Comment by glebashnik** [comment]
In readme, "weakAnd" needs a link. Also call  it "weakAnd" or "Weak AND".

File: rag-blueprint/README.md [hunk 2]
```diff
@@ -285,3 +288,106 @@ Intercept                     : -3.5974
...
+$ vespa query \
+ --timeout 60 \
+ --header="X-LLM-API-KEY:<your-api-key>" \
+ query="what are key points learned for finetuning llms?" \
+ queryProfile=rag \
+ ranking=second-with-gbdt
+</pre>
+
+Congratulations! You have now created a RAG application that can scale to billions of documents and thousands of queries per second, while still delivering state-of-the-art quality.
+What will you build?
```
Comment (glebashnik) [inline rag-blueprint/README.md:393]: Not done :)
Now we have our models. 
Running ranking evaluation is the next step.

**Review by glebashnik — APPROVED** [review]
Looks good!
See comments.

**Comment by glebashnik** [comment]
README requires a fair amount of background knowledge to understand. Would be great to add more links to our docs and introductory material related to infromation retrieval evaluation and learning to rank.

File: rag-blueprint/README.md [hunk 3]
```diff
@@ -241,12 +242,12 @@ We can see that all queries match all relevant documents, which is expected, sin
...
+### 2. First-phase ranking
 
-With our match-phase evaluation done, we can move on to the ranking phase. 
+With our match-phase evaluation done, we can move on to the ranking phase.
 We will start by collecting some training data for a handpicked set of features, which we will combine into a (cheap) linear first-phase ranking expression.
 
-#### Collect rank features
+### Collect matchfeatures
 
 In the rank-profile [`collect-training-data`](TODO), you can see we have createad both text-matching features (bm25), semantic similarity (embedding closeness), as well as document-level and chunk-level features. These are not normalized to the same range, which mean that we should learn the relationship (coefficients) between them.
```
Comment (radu-gheorghe) [inline rag-blueprint/README.md:252]: I know this doesn't have to do with this specific PR, but I think this link can be filled out now, no?

Comment (radu-gheorghe) [reply rag-blueprint/README.md:252]: I'm referring to the `collect-training-data` rank profile (which looks very nice, btw).

File: rag-blueprint/README.md (hunk 2 above)
Comment (thomasht86) [reply rag-blueprint/README.md:393]: Very good point - will be in next iteration!

File: rag-blueprint/README.md (hunk 3 above)
Comment (thomasht86) [reply rag-blueprint/README.md:252]: yes - good catch

**Comment by thomasht86** [comment]
Thanks for good review - will address the comments in the next iteration
